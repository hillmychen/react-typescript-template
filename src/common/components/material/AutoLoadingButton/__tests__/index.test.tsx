import React from 'react'
import { render, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { AutoLoadingButton } from '../index'
import userEvent from '@testing-library/user-event'

it('AutoLoadingButton click', () => {
  const onClick = jest.fn().mockImplementation(() => Promise.resolve())
  const { asFragment, getByRole } = render(<AutoLoadingButton onClick={onClick}>标题</AutoLoadingButton>)
  userEvent.click(getByRole('button'))
  expect(onClick).toBeCalled()
  expect(asFragment()).toMatchSnapshot()
})

it('AutoLoadingButton no Loading', async () => {
  const onClick = jest.fn().mockImplementation(() => Promise.resolve())
  const { asFragment, getByRole } = render(<AutoLoadingButton onClick={onClick}>按钮</AutoLoadingButton>)
  await waitFor(() => userEvent.click(getByRole('button')))
  expect(onClick).toBeCalled()
  expect(asFragment()).toMatchSnapshot()
})
