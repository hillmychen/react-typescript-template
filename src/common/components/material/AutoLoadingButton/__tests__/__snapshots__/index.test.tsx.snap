// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoLoadingButton click 1`] = `
<DocumentFragment>
  <button
    ant-click-animating-without-extra-node="false"
    class="ant-btn"
    type="button"
  >
    <span>
      标 题
    </span>
  </button>
</DocumentFragment>
`;

exports[`AutoLoadingButton no Loading 1`] = `
<DocumentFragment>
  <button
    ant-click-animating-without-extra-node="true"
    class="ant-btn"
    type="button"
  >
    <span>
      按 钮
    </span>
  </button>
</DocumentFragment>
`;
